### PR TITLE
HARP-6160: Improve the classification of outer/inner rings.

### PR DESCRIPTION
**Pull request by robertoraggi**
Use the winding of the 1st ring of a polygon feature
to distinguish between inner and outer rings.

Signed-off-by: Roberto Raggi <roberto.raggi@here.com>

### CODE REVIEW

File: @here/harp-omv-datasource/lib/OmvDecodedTileEmitter.ts [hunk 1]
```diff
@@ -837,9 +837,13 @@ export class OmvDecodedTileEmitter implements IOmvEmitter {
...
 
                 // Repeat the process for all the inner rings (holes).
+                const outerRingWinding = ring.winding;
                 const holes: number[] = [];
-                for (; ringIndex < rings.length && rings[ringIndex].isInnerRing; ++ringIndex) {
-                    const current = rings[ringIndex];
+
+                // Iterate over the inner rings. The inner rings have the oppositve winding
+                // of the outer rings.
+                while (ringIndex < rings.length && rings[ringIndex].winding !== outerRingWinding) {
```
Comment (zbigg) [inline @here/harp-omv-datasource/lib/OmvDecodedTileEmitter.ts]: I'm worried, that this loop stops on first "subring" that has same winding as ultimate parent ring, whereas old loop processed all rings. Is this intended ?

File: @here/harp-omv-datasource/lib/OmvDecoder.ts [hunk 2]
```diff
@@ -70,11 +70,7 @@ export class Ring {
         readonly contourOutlines?: boolean[],
         readonly points: number[] = contour
     ) {
-        this.isOuterRing = this.area() < 0;
-    }
-
-    get isInnerRing(): boolean {
-        return !this.isOuterRing;
+        this.winding = Math.sign(this.area());
```
Comment (ninok) [inline @here/harp-omv-datasource/lib/OmvDecoder.ts]: What about 0? The winding will be 0 as well. Not sure if the code will handle it properly.

File: @here/harp-omv-datasource/lib/OmvDecodedTileEmitter.ts (hunk 1 above)
Comment (zbigg) [reply @here/harp-omv-datasource/lib/OmvDecodedTileEmitter.ts]: Nevermind, outerloop takes care of processing remaining "outer" rings

File: @here/harp-omv-datasource/lib/OmvDecoder.ts (hunk 2 above)
Comment (robertoraggi) [reply @here/harp-omv-datasource/lib/OmvDecoder.ts]: that should not really happen, unless the data is broken.

Comment (robertoraggi) [reply @here/harp-omv-datasource/lib/OmvDecoder.ts]: i changed the condition to cover this case

**Pull request merged**